compiler/pipes/collect-main-edges.cpp

```diff
@@ -283,6 +283,7 @@ void CollectMainEdgesPass::on_func_call(VertexAdaptor<op_func_call> call) {
   for (int i = 0; i < call->args().size(); ++i) {
     VertexPtr arg = call->args()[i];
     auto param = function_params[i].as<op_func_param>();
+    bool is_cast_param = !stage::get_function()->file_id->is_strict_types && param->is_cast_param;
```

Comment:
Please, avoid using `stage::get_function()`, as it's not clear, what function we are talking about.
Here we should use `function` (a variable above), I suppose

Reply:
Maybe I need to access current file directly, without going through the function.

If we would use `function` variable above, we would make `strict_types` checked based on the declarations, but we want to apply it on the caller side. So if `f()` is declared in strict file, but called from the non-strict file, there should be no error. In that case, `f` is a called `function`, so it's file is a declaration file.

Reply:
May be `current_function` would be better in this case? 🤔

Reply:
> May be current_function would be better in this case? thinking

It is. But all methods of CollectMainEdgesPass are static. So in order to access it, we need to make them non-static, presumably.

Reply:
I used `current_function` in `preprocess-function.cpp` though.